Charles/add piuparts and uploaders

charles2910 opened the pull request:
Main thing here is the addition of piuparts test 81af25e5e8de529bdc72b4d358a48b5279c24c87. Then a revamp in the workflow  was done to bring it on par with our current setup from weston. And finally some small additions to build correctly locally were done.

Commit:
d/control: add uploaders field

Commit:
github-workflows: deb-builder: run piuparts test after build

Related-to: TCCP-706
Signed-off-by: Carlos Henrique Lima Melara <carlos.melara@toradex.com>

Commit:
github-workflows: deb-builder: build packages natively for arm and amd64

Use extra-docker-args to pass the correct platform to build natively the
packages using qemu syscall emulation.

Signed-off-by: Carlos Henrique Lima Melara <carlos.melara@toradex.com>

Commit:
github-workflows: deb-builder: bump checkout action to v4

Signed-off-by: Carlos Henrique Lima Melara <carlos.melara@toradex.com>

Commit:
github-workflows: deb-builder: use upstream build-deb action

We previously used a fork of build-deb, but upstream has picked up our changes
so we should use it which is maintained and developed. Also took the chance to
streamline the options passed to the action, removing unnecessary ones.

Signed-off-by: Carlos Henrique Lima Melara <carlos.melara@toradex.com>

Commit:
github-workflows: deb-builder: publish build artifacts

Signed-off-by: Carlos Henrique Lima Melara <carlos.melara@toradex.com>

leonheldattoradex commented on debian/source/options:
```diff
@@ -0,0 +1,2 @@
+# ignore github workflows stuff
```
Can you add a short note about why this is necessary in the commit message?

charles2910 replied:
Added more info in https://github.com/torizon/neofetch-deb/pull/1/commits/0ca762adb9567194d81a879e913babbc2b325cd2

leonheldattoradex reviewed (approved):
Looks good!

Commit:
d/s/options: make dpkg-source ignore our github workflow folder

Packaging modifications should only happen inside the debian/ directory,
but github doesn't support adding workflows outside .github/ so we need
to tell debian packageing tools to ignore this change introduced by us.

Signed-off-by: Carlos Henrique Lima Melara <carlos.melara@toradex.com>
